sympy/solvers/ode/single.py

```diff
@@ -186,10 +186,28 @@ class SingleODESolver:
     # Cache whether or not the equation has matched the method
     _matched = None  # type: Optional[bool]
 
+    # The order(s) of ODE that subclass can solve
+    _order = None  # type: int, list or string
+
     def __init__(self, ode_problem):
         self.ode_problem = ode_problem
 
     def matches(self) -> bool:
+        if type(self._order) == str:
+            # If the solver is not speific to any order
```

Comment:
I think we can simply this. Let's just make it so that `order` is either `None` or a tuple of orders.

Then the check is just
```python
if self.order is not None and self.ode_problem.order not in self.order:
    self._matched = False
    return False
```
It can be called `order` rather than `_order`.